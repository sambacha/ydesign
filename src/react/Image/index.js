import React from 'react';
import PropTypes from 'prop-types';
import Box from '../Box';

const Image = React.forwardRef((props, ref) => <Box ref={ref} {...props} />);

Image.defaultProps = {
  as: 'img',
  maxWidth: '100%',
  height: 'auto',
};

Image.propTypes = {
  ...Box.propTypes,
  theme: PropTypes.object,
  /**
   * Sets alt attribute on image for accessibility
   */
  alt: PropTypes.string,
  /**
   * Sets the src attribute of the img element
   */
  src: PropTypes.string.isRequired,
};

Image.displayName = 'Image';

export default Image;
