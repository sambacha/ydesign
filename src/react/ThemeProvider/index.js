import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
