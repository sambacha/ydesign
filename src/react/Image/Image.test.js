import Image from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Image component sanity', () => {
  it('has name', () => {
    expect(Image.displayName).toBe('Image');
  });

  it('matches default snapshot', () => {
    const component = render(<Image src="test.jpg" />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Image src="test.jpg" />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
