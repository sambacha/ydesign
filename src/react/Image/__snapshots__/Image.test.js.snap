// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  max-width: 100%;
  height: auto;
}

<img
  class="c0"
  height="auto"
  src="test.jpg"
/>
`;

exports[`Image component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  max-width: 100%;
  height: auto;
}

<img
  class="c0"
  height="auto"
  src="test.jpg"
/>
`;
